packages/Search/src/DTK_DistributedSearchTree_def.hpp

```diff
@@ -45,6 +45,18 @@ DistributedSearchTree<DeviceType>::DistributedSearchTree(
     _distributed_tree = std::make_shared<BVH<DeviceType>>( boxes );
 }
 
+template <typename DeviceType>
+typename DistributedSearchTree<DeviceType>::SizeType
+DistributedSearchTree<DeviceType>::size() const
```

Comment:
Why is this function `const` if the only variable it uses is `mutable`?

Reply:
Made more sense for the interface that the function is const since it does not change the bounding volume hierarchy.  The fact we cache the size for performance reasons is an implementation details that does not need to be exposed to the user IMO.

Reply:
That said, if we compute _size in the constructor, then it will not need to be declared mutable. I would argue that we don't actually need to compute it unless the user asks for it.

Reply:
+1 for lazy evaluation

Reply:
If we want to keep the function const, we should change the implementation. If the function is const it implies that the function is thread-safe. This is not the case with the current implementation unless `Teuchos::reduceAll` is thread-safe. We should guard the function with a mutex or use a temporary variable and an atomic assignment.

Reply:
@Rombur is right, we need to block the region with atomics/mutexes.